### PR TITLE
added preprocessing flag to guard stacktrace, turned off by default

### PR DESCRIPTION
**Pull request by mmorale3**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Added ENABLE_STACKTRACE to guard the use of boost/stacktrace in AFQMC code. Since this is only used for development, it needs to be turned on manually for now. In the future, if needed, we could add controls from cmake. 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix

### Does this introduce a breaking change?

- Yes

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/AFQMC/config.0.h [hunk 1]
```diff
@@ -27,12 +27,18 @@
...
 
 // guard with directive that checks if boost version is >=1.65
-#include <boost/version.hpp>
-#if BOOST_VERSION >= 106500
-#include <boost/stacktrace.hpp>
-#define print_stacktrace std::cout << boost::stacktrace::stacktrace();
+// uncomment to enable stacktrace
+//#define ENABLE_STACKTRACE
+#if defined(ENABLE_STACKTRACE)
+ #include <boost/version.hpp>
```
Comment (ye-luo) [inline src/AFQMC/config.0.h:33]: Will putting `#include <boost/version.hpp>` inside ENABLE_STACKTRACE potentially causes problems for multi?
How about
```
#if BOOST_VERSION >= 106500 && defined(ENABLE_STACKTRACE)
```

Comment (mmorale3) [reply src/AFQMC/config.0.h:33]: yeah, that's better

**Comment by mmorale3** [comment]
I would rather keep it like this for now, since I will remove this when we move to c++17.

**Comment by PDoakORNL** [comment]
Either its useful enough to go into the repo or it isn't.  I've been blocked multiple times from adding tooling hooks to the repo with support in the cmake and documentation.

**Comment by mmorale3** [comment]
added ENABLE_STACKTRACE to cmake

**Pull request merged**